docs/source/machine_translation.rst

```diff
@@ -0,0 +1,255 @@
+differences from the main Post_editing tutorial
+
+1 - Datasets
+************
+
+For training, we prepare two datasets. Since we will be using BPE, we need to
+ define the preprocessor. The configuration of the datasets looks like this::
+
+  [train_data]
+  class=config.utils.dataset_from_files
+  s_source=exp-nm-mt/data/train/Batch1a_en.txt.gz
+  s_target=exp-nm-mt/data/train/Batch1a_cs.txt.gz
+  preprocessor=<bpe_preprocess>
+
+  [val_data]
+  class=config.utils.dataset_from_files
+  s_source=exp-nm-mt/data/dev/Batch2a_en.txt.gz
+  s_target=exp-nm-mt/data/dev/Batch2a_cs.txt.gz
+  preprocessor=<bpe_preprocess>
+
+2 - Preprocessor and postprocessor
+**********************************
+
+Wee need to tell the Neural Monkey how it should handle preprocessing and
+postprocessing due to the BPR::
+
+  [bpe_preprocess]
+  class=processors.bpe.BPEPreprocessor
+  merge_file=exp-nm-mt/data/merge_file.bpe
+
+  [bpe_postprocess]
+  class=processors.bpe.BPEPostprocessor
+
+
+3 - Vocabularies
+****************
+
+For both encoder and decoder we will use shared vocabulary created from BPE
+merges::
+
+  [shared_vocabulary]
+  class=vocabulary.from_bpe
+  path=exp-nm-mt/data/merge_file.bpe
+
+4 - Encoder and Decoder
+************
+
+The encoder and decored is similar to Post-Editing ones::
+
+  [encoder]
+  class=encoders.sentence_encoder.SentenceEncoder
+  name=sentence_encoder
+  rnn_size=300
+  max_input_len=50
+  embedding_size=300
+  dropout_keep_prob=0.8
+  attention_type=decoding_function.Attention
+  data_id=source
+  vocabulary=<shared_vocabulary>
+
+  [decoder]
+  class=decoders.decoder.Decoder
+  name=decoder
+  encoders=[<encoder>]
+  rnn_size=256
+  embedding_size=300
+  use_attention=True
+  dropout_keep_prob=0.8
+  data_id=target
+  vocabulary=<shared_vocabulary>
+  max_output_len=50
+
+
+5 - Training sections
+**********************
+
+Following sections are described more in detail in Post-editing task::
+
+  [trainer]
+  class=trainers.cross_entropy_trainer.CrossEntropyTrainer
+  decoders=[<decoder>]
+  l2_weight=1.0e-8
+
+  [runner]
+  class=runners.runner.GreedyRunner
+  decoder=<decoder>
+  output_series=series_named_greedy
+
+  [bleu]
+  class=evaluators.bleu.BLEUEvaluator
+  name=BLEU-4
+
+  [ter]
+  class=evaluators.edit_distance.EditDistance
+  name=TER
+
+  [tf_manager]
+  class=tf_manager.TensorFlowManager
+  num_threads=4
+  num_sessions=1
+
+
+As for the main configuration section do not forget to add BPE postprocessing::
+
+  [main]
+  name=machine translation
+  output=exp-nm-mt/out-example-translation
+  runners=[<runner>]
+  tf_manager=<tf_manager>
+  trainer=<trainer>
+  train_dataset=<train_data>
+  val_dataset=<val_data>
+  evaluation=[(series_named_greedy,target,<bleu>), (series_named_greedy,target,
+  <ter>)]
+  minimize=True
+  batch_size=128
+  runners_batch_size=256
+  epochs=10
+  validation_period=1000
+  logging_period=20
+  save_n_best=3
+  postprocess=<bpe_postprocess>
```

Comment:
sem se dávat nemá, protože se má dávat do runneru. @@ to odstraňuje, a proto to ve výstupu neni.